src/Engine/RulesComponent.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Engine\RulesComponent.
+ */
+
+namespace Drupal\rules\Engine;
+
+use Drupal\Core\Plugin\Context\ContextDefinitionInterface;
+
+/**
+ * Handles executable Rules components.
+ */
+class RulesComponent {
+
+  /**
+   * The rules execution state.
+   *
+   * @var \Drupal\rules\Engine\RulesStateInterface
+   */
+  protected $state;
+
+  /**
+   * Definitions for the context used by the component.
+   *
+   * @var \Drupal\rules\Context\ContextDefinitionInterface[]
+   */
+  protected $contextDefinitions;
+
+  /**
+   * Definitions for the context provided by the component.
+   *
+   * @var \Drupal\rules\Context\ContextDefinitionInterface[]
+   */
+  protected $providedContextDefinitions;
+
+  /**
+   * The typed data manager.
+   *
+   * @var \Drupal\Core\TypedData\TypedDataManagerInterface
+   */
+  protected $typedDataManager;
+
+  /**
+   * The expression.
+   *
+   * @var \Drupal\rules\Engine\ExpressionInterface
+   */
+  protected $expression;
+
+  /**
+   * Constructs the object.
+   *
+   * @param \Drupal\rules\Engine\ExpressionInterface $expression
+   *   The expression of the component.
+   *
+   * @return static
+   */
+  public static function create(ExpressionInterface $expression) {
+    return new static($expression);
+  }
+
+  /**
+   * Constructs the object.
+   *
+   * @param \Drupal\rules\Engine\ExpressionInterface $expression
+   *   The expression of the component.
+   */
+  protected function __construct($expression) {
+    $this->state = RulesState::create();
+    $this->typedDataManager = \Drupal::typedDataManager();
```

Comment:
meh, no dependency injection. Can we use a method getTypedDataManager() that calls out globally if the typedDataManager class variable is not set? Then we can also have a setTypedDataManager() method that allows setter injection. Better than nothing.